lib/ProductOpener/Redis.pm

```diff
@@ -0,0 +1,50 @@
+
+package ProductOpener::Redis;
+
+use ProductOpener::PerlStandards;
+use Exporter    qw< import >;
+
+BEGIN
+{
+	use vars       qw(@ISA @EXPORT_OK %EXPORT_TAGS);
+	@EXPORT_OK = qw(
+		&push_to_search_service
+
+		);    # symbols to export on request
+	%EXPORT_TAGS = (all => [@EXPORT_OK]);
+}
+
+use vars @EXPORT_OK ;
+
+use Log::Any qw($log);
+use ProductOpener::Config2;
+use Redis::Client;
+
+sub init_redis() {
+	if ($ProductOpener::Config2::redis_url eq "")
+	{
+		$log->warn("Redis URL not provided for search indexing", { error => $@ }) if $log->is_warn();
+		return undef;
+	}
+	eval {
+		my $redis_client = Redis::Client->new(host => $ProductOpener::Config2::redis_url);
+		return $redis_client;
+	
+	};
+	if ($@) {
+		$log->warn("Error connecting to Redis", { error => $@ }) if $log->is_warn();
+	}
+	return undef;
+}
+
+my $redis_client = init_redis();
```

Comment:
@alexgarel @oalders - Hey there, since I restructured Redis to be a standalone file, this is no longer working. The problem is that the `$redis_client` variable is not getting set, and then isn't visible inside `push_to_search_service`. I am relearning Perl, so the solution is likely obvious.

From Googling I've tried a few things like:
- Using `our`, but this results in other unrelated errors, and I don't see it used in our codebase
- Putting in `use vars`, `use names`, etc.

Curious what I'm missing, and what the right way to do this is :)

Reply:
Also cc @stephanegigandet incase you have any idea

Reply:
Hi @simonj2 , this was difficult to debug, but I think the issue is the "return $redis_client;" which is inside the eval block: since it's in the eval, it's not returned to the caller of the init_redis() function. So you could do something like this:

```
sub init_redis() {

	$log->debug("init_redis", { redis_url_env => $ENV{REDIS_URL}, redis_url => $ProductOpener::Config2::redis_url } ) if $log->is_debug();
	
	$log->warn("REDIS_URL env", {redis_url => $ENV{REDIS_URL} }) if $log->is_warn();
	if ($ProductOpener::Config2::redis_url eq "")
	{
		$log->warn("Redis URL not provided for search indexing", { error => $@ }) if $log->is_warn();
		return undef;
	}
	my $redis_client;
	eval {
		$redis_client = Redis::Client->new(host => $ProductOpener::Config2::redis_url);
	};
	if ($@) {
		$log->warn("Error connecting to Redis", { error => $@ }) if $log->is_warn();
	}
	else {
		return $redis_client;
	}
	return undef;
}
```

But since you have my $redis_client = init_redis() after, I woud just declare "my $redis_client;" at the top of  the file (before init_redis()), and init_redis() could just set it directly, without returning it.

Reply:
Oh amazing, thank you. I'll give this a shot. Appreciate your help!

Reply:
This worked - thank you!